enarx-keep/src/backend/kvm/vm.rs

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod builder;
+mod cpu;
+mod mem;
+mod syscall;
+mod x86_64;
+
+pub use builder::Builder;
+use cpu::Cpu;
+use mem::Region;
+
+use crate::backend::{Keep, Thread};
+
+use kvm_ioctls::{Kvm, VmFd};
+
+use std::cell::RefCell;
+use std::io::Result;
+use std::sync::Arc;
+
+pub struct VirtualMachine {
+    _kvm: Kvm,
+    _fd: VmFd,
+    _address_space: Region,
+    deferred_cpu: RefCell<Option<Cpu>>,
```

Comment:
Is `RefCell` the right thing here?

Reply:
`Keep::add_thread` doesn't provide a mutable reference to the keep (and by extension: this field). RefCell was the only trick I could think of for the interior mutability here.

Reply:
@npmccallum Do you want this PR to update that interface so that `Keep::add_thread` provides mutability? I chose not to initially because I didn't want to make any invasive changes for something that may be just a 1-off for my case. I'll be the first to admit that my foresight is probably more shortsighted here.

Reply:
Yes, I think that's the better option. I think the best explanation for why it isn't mutable is an oversight on my part.

We probably need to `Arc<Mutex<dyn Keep>>` so that any thread can handle the new thread creation process via keeping a reference to the keep and locking it when necessary.

Reply:
Mutex => RwLock

Reply:
Addressed in latest revision in a separate commit. The compiler didn't let me use `Arc<RwLock<dyn Keep>>` as a receiver like the trait function currently does with `Arc<dyn Keep>`. In my research I found that many solutions present an interface that looks single threaded and relies on Rust's compile-time checks to ensure the caller wraps their use in types that are thread safe if the data is sent across to another thread, so that's the approach that I took here as well.